src/App.tsx

```diff
@@ -1,27 +1,72 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.scss';
 import { GoodsList } from './GoodsList';
+import { Good } from './types/Good';
 
-// import { getAll, get5First, getRed } from './api/goods';
-// or
-// import * as goodsAPI from './api/goods';
+import { getAll, get5First, getRedGoods } from './api/goods';
 
-export const App: React.FC = () => (
-  <div className="App">
-    <h1>Dynamic list of Goods</h1>
+export const App: React.FC = () => {
+  const [goods, setGoods] = useState<Good[]>([]);
 
-    <button type="button" data-cy="all-button">
-      Load all goods
-    </button>
+  const handlerAll = () => {
+    getAll()
+      .then(todosFromServer => {
+        setGoods(todosFromServer);
+      })
+      .catch(() => {
+        setGoods([]);
+      });
+  };
 
-    <button type="button" data-cy="first-five-button">
-      Load 5 first goods
-    </button>
+  const handler5First = () => {
+    get5First()
+      .then(todosFromServer => {
+        setGoods(todosFromServer);
+      })
+      .catch(() => {
+        setGoods([]);
+      });
+  };
 
-    <button type="button" data-cy="red-button">
-      Load red goods
-    </button>
+  const handlerRed = () => {
+    getRedGoods()
+      .then(todosFromServer => {
+        setGoods(todosFromServer);
+      })
+      .catch(() => {
+        setGoods([]);
+      });
+  };
 
-    <GoodsList goods={[]} />
-  </div>
-);
+  return (
+    <div className="App">
+      <h1>Dynamic list of Goods</h1>
+
+      <button
+        type="button"
+        data-cy="all-button"
+        onClick={() => handlerAll()}
+      >
```

Comment:
Consider using `async/await` instead of `.then()` and `.catch()` for better readability and error handling. 🤔